lib/python-beta/src/bailo/core/client.py

```diff
@@ -315,6 +315,31 @@ def get_download_file(
         shutil.copyfileobj(req.raw, buffer)
         return file_id
 
+    def get_download_file_token(
+        self,
+        model_id: str,
+        file_id: str,
+        buffer: BytesIO,
+    ):
+        """
+        Downloads a specific file, using an API token.
+
+        :param model_id: Unique model ID
+        :param file_id: Unique file ID
+        :param buffer: BytesIO object for bailo to write to
+        :return: The unique file ID
+
+        ..note:: This method requires the TokenAgent agent.
+        """
+        if isinstance(self.agent, TokenAgent):
```

Comment:
`isinstance` is a built in function that considers inheritance eg `isinstance(Agent(), TokenAgent)` will be `True`. Instead you want something like `type(self.agent) == TokenAgent`